tests/pkg/mcufont/main.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2023 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the MCUFont package
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "mcufont.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#define PIXBUF_NUM_ROWS     12
+#define PIXBUF_NUM_COLS     16
+#define PIXBUF_ALPHA_THRESHOLD  80
+#define MCUFONT_TEST_FONT mf_rlefont_DejaVuSans12bw
+
+extern const struct mf_rlefont_s MCUFONT_TEST_FONT;
+
+static const char input[] = "Hello, RIOT!";
+static uint8_t pixbuf[PIXBUF_NUM_ROWS][PIXBUF_NUM_COLS] = { 0 };
+
+static void pixel_callback(int16_t x, int16_t y, uint8_t count, uint8_t alpha, void *state)
+{
+    (void)state;
+    DEBUG("[draw] x: %"PRIi16", y: %"PRIi16", count: %u, color: %u\n", x, y, count, alpha);
+    if (y >= PIXBUF_NUM_ROWS || ((x + count) >= (PIXBUF_NUM_COLS * 8))) {
+        return;
+    }
+    while (count--) {
+        uint8_t bitpos = x % 8;
+        uint8_t bytepos = x / 8;
+        if (alpha > PIXBUF_ALPHA_THRESHOLD) {
+            pixbuf[y][bytepos] |= (1 << (7 - bitpos));
+        }
+        x++;
+    }
+}
+
+static void print_pixbuf(void)
+{
+    for (size_t y = 0; y < PIXBUF_NUM_ROWS; y++) {
+        for (size_t x = 0; x < PIXBUF_NUM_COLS; x++) {
+            for (int bitpos = 7; bitpos >= 0; bitpos--) {
+                if (pixbuf[y][x] & (1 << bitpos)) {
+                    printf("█");
+                }
+                else {
+                    printf(" ");
+                }
+            }
+        }
+        puts("|"); /* Adds a visual pixbuf end marker and a newline */
+    }
+}
+
+static uint8_t char_callback(int16_t x0, int16_t y0, mf_char character, void *state)
+{
+    return mf_render_character(&MCUFONT_TEST_FONT.font, x0, y0, character, &pixel_callback, state);
+}
+
+int main(void)
+{
+    puts("mcufont package test application");
+    mf_render_aligned(
+            &MCUFONT_TEST_FONT.font,
+            0, 0,
+            MF_ALIGN_LEFT,
+            input, strlen(input),
+            &char_callback, NULL);
+
+    print_pixbuf();
```

Comment:
It would be nice if the message was also displayed on a screen for boards that provides one. The qrcode package test application does that.

Reply:
Added something that works (not something great though):

![image](https://github.com/RIOT-OS/RIOT/assets/5160052/d55d8eb2-edfc-4517-976b-3800bd579a58)